src/Medidata.ZipkinTracer.Core/SpanCollector.cs

```diff
@@ -13,13 +13,20 @@ public class SpanCollector
         internal SpanProcessor spanProcessor;
 
         private static SpanCollector instance;
+        private static readonly object syncObj = new object();
 
         public static SpanCollector GetInstance(Uri uri, uint maxProcessorBatchSize, ILog logger)
         {
             if (instance == null)
             {
-                instance = new SpanCollector(uri, maxProcessorBatchSize, logger);
-                instance.Start();
+                lock (syncObj)
+                {
+                    if (instance == null)
```

Comment:
How do you guarantee that two threads haven't already go through this condition before the instance was effectively created?